src/runtime/extensiontype.cs

```diff
@@ -76,10 +76,10 @@ protected virtual void Clear()
         /// <summary>
         /// Type __setattr__ implementation.
         /// </summary>
-        public static int tp_setattro(IntPtr ob, IntPtr key, IntPtr val)
+        public static int tp_setattro(BorrowedReference ob, BorrowedReference key, BorrowedReference val)
```

Comment:
Neat, I did not know this would work

Reply:
Dang. Judging by failures it did not. You'd need to construct a delegate type with the right signature at runtime using [Expression.GetFuncType](https://docs.microsoft.com/en-us/dotnet/api/system.linq.expressions.expression.getfunctype?redirectedfrom=MSDN&view=net-5.0#System_Linq_Expressions_Expression_GetFuncType_System_Type___)

BTW, in the future, we might make this faster by using [`[UnmanagedCallersOnly]`](https://docs.microsoft.com/en-us/dotnet/api/system.runtime.interopservices.unmanagedcallersonlyattribute) (.NET 5+)